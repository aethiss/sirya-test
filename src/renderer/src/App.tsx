import { ReactElement, useEffect, useState } from 'react'

import electronLogo from './assets/electron.svg'
import Config from './components/config/Config'
import { ipcRenderer } from 'electron'

const App = (): ReactElement => {
  const [salt, setSalt] = useState<string>('')
  console.log('ðŸš€ ~ salt =>', salt)

  useEffect(() => {
    ;(async (): Promise<void> => {
      const result = await ipcRenderer.invoke('openConfigFile')
      result && setSalt(result.salt)
    })()
  }, [])

  return (
    <>
      <img alt="logo" className="logo" src={electronLogo} />
      <Config salt={salt} />
    </>
  )
}

export default App
