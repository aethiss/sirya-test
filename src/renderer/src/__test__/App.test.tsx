import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import App from '../App'

// Mocks

describe('Rendering', () => {
  test('Renders the main page', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})

// describe('Functionality', () => {
//   test.only('If config file is loaded', async () => {
//     render(<App />)
//     expect(ipcRenderer.invoke).toHaveBeenCalled()
//   })
// })
