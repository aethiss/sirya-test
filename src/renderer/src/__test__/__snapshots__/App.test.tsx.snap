// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders the main page 1`] = `
<DocumentFragment>
  <img
    alt="logo"
    class="logo"
    src="[object Object]"
  />
  <div>
    <button
      data-testid="config-button"
    >
      Select config file
    </button>
  </div>
</DocumentFragment>
`;
